test/wampy-json-test.js

```diff
@@ -32,13 +32,46 @@ describe('Wampy.js [with JSON serializer]', function () {
         it('allows to connect on instantiation if all required options specified', function (done) {
             let wampy = new Wampy(routerUrl, {
                 realm: 'AppRealm',
-                onConnect: done,
+                onConnect: function () {
+                    done();
+                },
                 ws: WebSocketModule.WebSocket,
                 serializer: new JsonSerializer()
             });
             expect(wampy).to.be.an('object');
         });
 
+        it('passes welcome details to onConnect() callback', function (done) {
+            let wampy = new Wampy(routerUrl, {
+                realm: 'AppRealm',
+                onConnect: function (welcomeDetails) {
+                    expect(welcomeDetails).to.be.an('object');
+                    done();
+                },
+                ws: WebSocketModule.WebSocket,
+                serializer: new JsonSerializer()
+            })
+        });
+
+        it('passes welcome details to onReconnectSuccess() callback', function (done) {
```

Comment:
@KSDaemon for some reason, this test isn't passing. Can you check, please?

Reply:
Yep, i'll take a look